dali/operators/imgcodec/util/convert_test.cc

```diff
@@ -0,0 +1,510 @@
+ */
+template <typename T, int ndim>
+void Crop(const TensorView<StorageCPU, T, ndim> &output,
+          const TensorView<StorageCPU, const T, ndim> &input, const ROI &requested_roi,
+          const TensorLayout &layout = "HWC") {
+  auto roi = ExtendRoi(requested_roi, input.shape, layout);
+
+  static_assert(ndim >= 0, "expected static ndim");
+  ASSERT_TRUE(output.shape == roi.shape());  // output should have the desired shape
+
+  kernels::SliceCPU<T, T, ndim> kernel;
+  kernels::SliceArgs<T, ndim> args;
+  args.anchor = roi.begin;
+  args.shape = roi.shape();
+  kernels::KernelContext ctx;
+  // no need to run Setup (we already know the output shape)
+  kernel.Run(ctx, output, input, args);
+}
+
+template <typename T, int ndim>
+Tensor<CPUBackend> Crop(const TensorView<StorageCPU, const T, ndim> &input,
+                        const ROI &requested_roi, const TensorLayout &layout = "HWC") {
+  auto roi = ExtendRoi(requested_roi, input.shape, layout);
+  auto num_dims = input.shape.sample_dim();
+  assert(roi.shape().sample_dim() == num_dims);
+  Tensor<CPUBackend> output;
+  output.Resize(roi.shape(), type2id<T>::value);
+
+  auto out_view = view<T, ndim>(output);
+  Crop(out_view, input, roi);
+
+  return output;
+}
+
+static Tensor<CPUBackend> Crop(const Tensor<CPUBackend> &input, const ROI &roi) {
+  int ndim = input.shape().sample_dim();
+  VALUE_SWITCH(ndim, Dims, (2, 3, 4), (
+    TYPE_SWITCH(input.type(), type2id, InputType, (IMGCODEC_TYPES, double), (
+      return Crop(view<const InputType, Dims>(input), roi, input.GetLayout());
+    ), DALI_FAIL(make_string("Unsupported type ", input.type())););  // NOLINT
+  ), DALI_FAIL(make_string("Unsupported number of dimensions: ", ndim)););  // NOLINT
+}
+
+/**
+ * @brief Checks if the image and the reference are equal
+ */
+template <typename T>
+void AssertEqual(const TensorView<StorageCPU, const T> &img,
+                 const TensorView<StorageCPU, const T> &ref) {
+  Check(img, ref);
+}
+
+/**
+ * @brief Checks if the image and the reference are equal after converting the reference
+ * with ConvertSatNorm
+ */
+template <typename T, typename RefType>
+void AssertEqualSatNorm(const TensorView<StorageCPU, const T> &img,
+                        const TensorView<StorageCPU, const RefType> &ref) {
+  Check(img, ref, EqualConvertSatNorm());
+}
+
+template <typename T>
+void AssertEqualSatNorm(const TensorView<StorageCPU, const T> &img, const Tensor<CPUBackend> &ref) {
+  TYPE_SWITCH(ref.type(), type2id, RefType, NUMPY_ALLOWED_TYPES, (
+    Check(img, view<const RefType>(ref), EqualConvertSatNorm());
+  ), DALI_FAIL(make_string("Unsupported reference type: ", ref.type())));  // NOLINT
+}
+
+static void AssertEqualSatNorm(const Tensor<CPUBackend> &img, const Tensor<CPUBackend> &ref) {
+  TYPE_SWITCH(img.type(), type2id, T, (IMGCODEC_TYPES), (
+    AssertEqualSatNorm<T>(view<const T>(img), ref);
+  ), DALI_FAIL(make_string("Unsupported type: ", img.type())));  // NOLINT
+}
```

Comment:
These are pretty generic utilities. Previously there was a "test_helper" file for those, now they end up buried in a .cc file, with little hope of reuse.